### PR TITLE
Update 1028-zh-tw.cfos.txt

### PR DESCRIPTION
**Pull request by LNDDYL**
Update translation.

### CODE REVIEW

File: locales.software/1028-zh-tw.cfos.txt [hunk 1]
```diff
@@ -134,7 +134,7 @@ start_product = "啟動 %p";
 info_port_in_use = "%p 要使用的連接埠 %? 正被其它程式使用。\n\n程式：%?\n路徑：%?\n\n要使用 %p，您必須結束該程式或是設定 %p 使用其它連接埠。";
 monitor_port_desc = "本程式持續監控 % 要使用的連接埠，若連接埠被其它程式使用，便會警告使用者。";
 logon_start_desc = "以下的使用者登入時，本任務就會啟動 %p。";
-show_home_dir = "顯示使用者家目錄";
+show_home_dir = "顯示使用者首目錄";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: 我想「家目錄」可能比較好一點。

File: locales.software/1028-zh-tw.cfos.txt [hunk 2]
```diff
@@ -309,7 +309,7 @@ delete = "刪除";
 aux_lines = "輔助線";
 higher_prog_help = "網路電話、或是遊戲";
 high_prog_help = "遊戲、串流多媒體、Telnet 及其他的互動程式";
-low_prog_help = "伺服器資料, FTP 資料";
+low_prog_help = "伺服器資料， FTP 資料";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: 這邊應該要使用頓號，並且去掉 FTP 前面多餘的空白。

File: locales.software/1028-zh-tw.cfos.txt [hunk 3]
```diff
@@ -357,7 +357,7 @@ last_3_months = "最近 3 個月";
 last_month = "上個月";
 last_12_months = "最近 12 個月";
 please_select = "請選擇";
-reset_day = "重置日期";
+reset_day = "重設日期";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: equally

File: locales.software/1028-zh-tw.cfos.txt [hunk 4]
```diff
@@ -775,12 +775,12 @@ err_no32 = "無法在 64 位元的作業系統安裝 32 位元的驅動程式。
 err_bad_keyfile = "授權檔案似乎已損毀或是無效，請聯絡我們以取得有效的授權檔案。";
 err_bad_setup = "%p 似乎不正確的安裝或是某些檔案被修改，請重新安裝 %p 後重試。";
 err_bad_signature = "密碼似乎不正確，或是另一個授權檔案使用的密碼。請輸入符合這個授權檔案的正確密碼。";
-err_gdiplus = "找不到 GDI+ 函式庫！\n\n若要繼續安裝，必需先安裝 GDI+ 函式庫，\n您可以下載微軟的平台軟體開發工具組取得 GDI+ 函式庫，\n請下載安裝檔案並解壓縮所有檔案至任一目錄，\n然後將解開目錄下 \\asms\\10\\msft\\windows\\gdiplus\n裏的所有 gdiplus.* 檔案，\n複製到 Windows\\system32 目錄裏。\n\n點擊「確定」打開微軟的相關下載頁面。\n點擊「取消」中止目前的安裝。";
+err_gdiplus = "找不到 GDI+ 函式庫！\n\n若要繼續安裝，必需先安裝 GDI+ 函式庫，\n您可以下載微軟的平台軟體開發工具組取得 GDI+ 函式庫，\n請下載安裝檔案並解壓縮所有檔案至任一目錄，\n然後將解開目錄下 \\asms\\10\\msft\\windows\\gdiplus\n裡的所有 gdiplus.* 檔案，\n複製到 Windows\\system32 目錄裡。\n\n點擊「確定」開啟微軟的相關下載頁面。\n點擊「取消」中止目前的安裝。";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: 「打開」與「開啟」我覺得是相等的，其餘 OK。

File: locales.software/1028-zh-tw.cfos.txt [hunk 5]
```diff
@@ -775,12 +775,12 @@ err_no32 = "無法在 64 位元的作業系統安裝 32 位元的驅動程式。
...
 err_winhttp = "找不到 WinHTTP 函式庫！\n\n若要繼續安裝，必需先安裝 WinHTTP 函式庫，\nWinHTTP 函式庫應該內建於您作業系統中，\n但是目前無法載入。";
 bad_manufacturer = "無法啟動安裝，因為這個產品只授權給特定的硬體！";
 cannot_install = "無法安裝";
 err_uninst_daemon = "無法移除服務。如果還有開啟 %p 的對話視窗，請先關閉這些視窗，否則請重新開機。";
-err_activate_10045 = "啟動授權碼時發生 HTTP 錯誤 10045，可能是您的 Winsock\n已損毀。要重置您的 Winsock，請開啟「命令提示字元」\n視窗，輸入「netsh winsock reset」，\n然後重新啟動您的系統，再試著啟動授權碼。";
+err_activate_10045 = "啟動授權碼時發生 HTTP 錯誤 10045，可能是您的 Winsock\n已損毀。要重設您的 Winsock，請開啟「命令提示字元」\n視窗，輸入「netsh winsock reset」，\n然後重新啟動您的系統，再試著啟動授權碼。";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: 「重置」與「重設」我覺得是相等的。

File: locales.software/1028-zh-tw.cfos.txt [hunk 6]
```diff
@@ -837,7 +837,7 @@ virtual_wifi_error_get_cons = "取得網路連線時發生錯誤。";
 virtual_wifi_ics_warning = "已為下列連線啟用網際網路連線共用:\n%?\n\n您是否想改用此網路:\n%?";
 virtual_wifi_error_enable_ics = "啟用網際網路連線共用時發生錯誤。";
 virtual_wifi_error_start = "啟動虛擬 Wi-Fi 時發生錯誤。";
-virtual_wifi_ok = "已成功啟用虛擬 Wi-Fi 與網際網路連線共用。\n\n註：若客戶端無法正常連線至虛擬 Wi-Fi，請停止並重新開啟虛擬 Wi-Fi。\n\n若客戶端可以連線至虛擬 Wi-Fi 但無法連線至網際網路，請移除並重新安裝 %p。";
+virtual_wifi_ok = "已成功啟用虛擬 Wi-Fi 與網際網路連線共用。\n\n註：若用戶端無法正常連線至虛擬 Wi-Fi，請停止並重新開啟虛擬 Wi-Fi。\n\n若用戶端可以連線至虛擬 Wi-Fi 但無法連線至網際網路，請移除並重新安裝 %p。";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: 「用戶端」與「客戶端」我覺得也是相等的。

File: locales.software/1028-zh-tw.cfos.txt [hunk 7]
```diff
@@ -901,13 +901,13 @@ actionwait = "連上網路之後，將執行您指定的行為。";
 discask = "確定要中斷此連線？";
 dial_warn = "允許撥打 %1 嗎?";
 / %1, %2 are Windows version, e.g. "Windows 95", "Windows XP"
-badvsn = "您使用的 %p 程式適用於 %1，\n但您的金鑰只能在 %2 使用。\n您可在以下網址訂購升級版：\nhttps://www.cfos.de/zh-tw/order/order.htm\n\n如果您剛購買了 %p 卻發現版本錯誤，請與我們聯繫：\n%url/contact/contact.htm";
+badvsn = "您使用的 %p 程式適用於 %1，\n但您的金鑰只能在 %2 使用。\n您可在以下網址訂購升級版：\nhttps://www.cfos.de/zh-tw/order/order.htm\n\n如果您剛購買了 %p 卻發現版本錯誤，請與我們聯絡：\n%url/contact/contact.htm";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: equally

File: locales.software/1028-zh-tw.cfos.txt [hunk 8]
```diff
@@ -901,13 +901,13 @@ actionwait = "連上網路之後，將執行您指定的行為。";
...
 
 active = "使用中";
 inactive = "未使用";
 disabled = "已停用";
 setup = "驅動程式安裝";
-no_write = "沒有 %p 檔案的寫入權限\n%1。\n\n請洽詢您的電腦管理員，讓您獲得 CFOS.INI 與 GLOBAL.INI 檔案的寫入和修改權限。\n";
+no_write = "沒有 %p 檔案的寫入權限\n%1。\n\n請洽詢您的電腦管理員，讓您取得 CFOS.INI 與 GLOBAL.INI 檔案的寫入和修改權限。\n";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: equally

File: locales.software/1028-zh-tw.cfos.txt [hunk 9]
```diff
@@ -982,11 +982,11 @@ dlg_med_isdn_box = "ISDN (CAPI 2.0)";
 dlg_med_dsl_box = "DSL（PPPoE 撥號連線）";
 dlg_med_areacode_txt = "區號：";
 dlg_med_msn_txt = "MSN（您的 ISDN 電話號碼）：";
-dlg_med_netbt_txt = "調整 TCP/IP 設定以便獲得最大效能，並取消 NetBIOS over TCP/IP（推薦）";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: equally

File: locales.software/1028-zh-tw.cfos.txt [hunk 10]
```diff
@@ -1035,7 +1035,7 @@ dlg_dun_header_dsl = "DSL 連線";
 dlg_dun_header_isdn = "ISDN 連線";
 dlg_dun_subheader = "請確定您已輸入完整及正確的資料，尤其是您的帳號與密碼。";
 dlg_dun_phone = "撥號號碼：";
-dlg_dun_line = "包括外線前綴號碼（例如用公司電話在撥打外線需先撥 0）（若需要）";
+dlg_dun_line = "包括外線首碼號碼（例如用公司電話在撥打外線需先撥 0）（若需要）";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: 我認為保持「前綴」比較好。

File: locales.software/1028-zh-tw.cfos.txt [hunk 11]
```diff
@@ -1063,12 +1063,12 @@ dlg_blk_block = "需要封鎖的號碼";
 dlg_blk_warn = "需要警告的號碼";
 dlg_blk_add = "新增";
 dlg_blk_del = "移除";
-dlg_blk_prefix = "前綴號碼";
+dlg_blk_prefix = "首碼號碼";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: same as above

File: locales.software/1028-zh-tw.cfos.txt [hunk 12]
```diff
@@ -1063,12 +1063,12 @@ dlg_blk_block = "需要封鎖的號碼";
...
 dlg_blk_ask_pref = "";
 dlg_blk_ras = "所有的撥號連線";
 dlg_blk_doublechk = "雙重檢測";
-dlg_blk_doublechk_help = "如果您想檢查包含與不含前綴號碼\n的部分，請勾選「雙重檢測」。\n通常只在與您的電腦連結的部分裝置需要，而另一部分\n又不需要撥號前綴時，才需要勾選此選項。\n";
-dlg_blk_legalinfo = "cFosWatch 與 %p 無法保護的情況：\n如果您同時安裝了 ISDN 卡與傳統類比數據機，\n這些裝置就還是可能撥打到需要被封鎖的昂貴電話號碼。\n如果這發生在沒有 %p 或撥號連線（例：有第三方程式\n透過自訂的方式建立語音或數據連線），就無法被\n%p 或 cFosWatch 偵測到。這種情況下，您必須\n直接拔掉數據機的插頭或關閉電源。目前也沒有辦法保護\n由未安裝　%p 或 cFosWatch 的電腦和傳統電話所撥出的\n連線。若您想要得到完整的保護，請直接向電信業者申請\n封鎖某些電話號碼，但這種服務可能也不是免費的。此外，\n我們無法為 %p 和/或 cFosWatch 的功能失效，或是作業系統\n與第三方軟體功能負責。若其他第三方軟體刻意停用或規避\n%p 和/或 cFosWatch 的功能時，恕本公司亦不承擔任何責任。\n\n本文也保存到 %p 資料夾的 0190.TXT 文件中。\n";
+dlg_blk_doublechk_help = "如果您想檢查包含與不含首碼號碼\n的部分，請勾選「雙重檢測」。\n通常只在與您的電腦連結的部分裝置需要，而另一部分\n又不需要撥號首碼時，才需要勾選此選項。\n";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: same as above

File: locales.software/1028-zh-tw.cfos.txt [hunk 13]
```diff
@@ -1106,11 +1106,11 @@ pr_inst_dun = "正在安裝 Windows 撥號連線…";
 pr_restart_point = "正在建立還原點…";
 
 / ISDN late install
-late_ask = "AOL 在您的電腦偵測到 ISDN 設備，您是否想安裝\nAOL 的 ISDN 支援程式 (%p OEM)？\n\n若您想立即安裝 ISDN 支援程式，請點擊「是」。\n\n若您想稍候再安裝 ISDN 支援程式，請點擊「否」。\n\n若您不想安裝 ISDN 支援程式，也不想再被詢問，請點擊「取消」。";
+late_ask = "AOL 在您的電腦偵測到 ISDN 裝置，您是否想安裝\nAOL 的 ISDN 支援程式 (%p OEM)？\n\n若您想立即安裝 ISDN 支援程式，請點擊「是」。\n\n若您想稍候再安裝 ISDN 支援程式，請點擊「否」。\n\n若您不想安裝 ISDN 支援程式，也不想再被詢問，請點擊「取消」。";
 late_instnever_confirm = "注意：您選擇了「取消」，將不會安裝 AOL 的支援程式。\n您有可能無法透過 ISDN 上網。\n如果您想透過 ISDN 連線至 AOL，我們則建議您安裝 ISDN 支援程式。\n\n您要安裝 ISDN 支援程式嗎？";
 
 / DSL late install
-late_dsl_ask = "AOL 在您的電腦偵測到 DSL 設備，您是否想安裝\nAOL 的 DSL 支援程式 (%p OEM)？\n\n若您想立即安裝 DSL 支援程式，請點擊「是」。\n\n若您想稍候再安裝 DSL 支援程式，請點擊「否」。\n\n若您不想安裝 DSL 支援程式，也不想再被詢問，請點擊「取消」。";
+late_dsl_ask = "AOL 在您的電腦偵測到 DSL 裝置，您是否想安裝\nAOL 的 DSL 支援程式 (%p OEM)？\n\n若您想立即安裝 DSL 支援程式，請點擊「是」。\n\n若您想稍候再安裝 DSL 支援程式，請點擊「否」。\n\n若您不想安裝 DSL 支援程式，也不想再被詢問，請點擊「取消」。";
```
Comment (petercpg) [inline locales.software/1028-zh-tw.cfos.txt]: equally

**Comment by petercpg** [comment]
Hi @LNDDYL，感謝送出 pull request。標示為 equally 的句子是我認為沒有不同的句子，我認為若兩句間的用詞若沒有差異則應該保留原譯法以尊重較早的譯者。歡迎再 follow-up 討論。

@ArmorDarks please don't merge this PR until we have agreed for the terms used. :)

**Comment by ArmorDarks** [comment]
@petercpg Sure. Thanks for taking care of that PR!

Just as side note, I think it's ok to change terminology if changes leads to better translation. But in such case all older translation would be better to be updated too.

Choice of better translation is completely up to you @petercpg, since I don't speak Chinese :)

**Comment by petercpg** [comment]
@ArmorDarks 
I just don't want to raise edit wars like changing between, say, "folder" and "directory" (Just an example, not a case here) 

I'm agree to the changes that I didn't leave a note on, but some terms changed are equal imo, so I'm trying to discuss with @LNDDYL to make the quality better..

**Comment by LNDDYL** [comment]
有台灣本土詞彙，應優先使用本土詞彙。再者，我是以Microsoft語言入口翻譯標準翻譯的。https://www.microsoft.com/Language/zh-TW/Default.aspx

**Comment by LNDDYL** [comment]
「,」怎麼可能是頓號，這是西文逗點好不好？！在中文裡面使用西文標點符號，難道不是很怪異嗎？

**Comment by petercpg** [comment]
@LNDDYL 有一些原本的翻譯的確是很奇怪的機器翻譯，我上次整批更新的時候還是有漏掉，所以像是你要改「裡」跟「裏」這種正體異體間的差異我沒有意見。但其他同等的用詞翻譯我覺得沒有修改的必要，不然哪一天又有人有不同意見又來改一次只是浪費大家時間而已。

另外微軟的翻譯未必就是好翻譯，除非是軟體介面之類需要照翻的用詞（例如「檔案總管」、「網路上的芳鄰」）外，我不覺得需要跟著翻譯。Microsoft Support 上自己很多機器翻譯的文章也沒照著語彙庫來翻，使用前請多斟酌。

至於 `show_home_dir` 翻出來之後該不該是頓號....

我想這應該很清楚？

**Comment by ArmorDarks** [comment]
@petercpg Sure, I understand your point.

Regarding Microsoft translations — indeed, sometime for some languages it's only partially reliable source. Best practice is to follow common among developers lexicon. Best tool to check is Google (or regional similar service) — if you have concern regarding translation of some term, you always can check it there. If chosen translation never encounters there in similar context to translatable string's one or very rare — probably, it's better to search for another common translation for that term.

I'm not sure is it relative to arised issue, since Google's translation is very inaccurate, but I hope that it will help at least a little.

**Comment by petercpg** [comment]
@ArmorDarks That's how I do when I met some technical terms, but for more common terms, I think Google's problem is that it's smart enough to detect what I am searching, but not accurate enough because of the fuzzy logic of language nature, especially when most Chinese articles on Microsoft Support since Vista-era are machine translated, and Microsoft's machine translation engine did not follow their lexicon.

The term usage of Chinese is not that strict so we can find different ways to describe some terms, say, the change proposed for entity `reset_day` is should we use "重置" or "重設" ? They both reflects term "reset" then why are you using A but not B? Google wouldn't tell which is better because they are both good. 

I believe this is a big headache for localizers over the world....

**Comment by ArmorDarks** [comment]
@petercpg I agree with your point. Problem for sure exists for every language. For some it's just easier to check best translation then for other. For common cyrillic and latin languages usually Google + Wikipedia + topics/discussions about relative themes on forums is best treatment, even when it comes to such ordinary words as `reset` — it's just enough to search for that word in similar context (phrase) to ensure that chosen form/variant is appropriate (or not). I guess it just doesn't work well for Chinese.

I'm also really surprised that Microsoft used machine translation. In Russian, for example, translation is quite accurate. In fact, it was Microsoft who populated a lot of now common translations for IT terms, which even didn't exist in Russian language before. In such situation relaying on Microsoft translation definitely not way to go.

Regarding different ways to describe one terms — that problem more then known personally for me, because Russian and Ukrainian languages has exactly similar problem. There is just tons of ways to say same things, not like in English.

Yes, one more point. Why I've mentioned Google. When localizing application, we have to think not only about proper translation, but also about searchability of used term. For example, let's imagine situation when we have term `Traffic Shaping` translated into some UnknownOtherLanguage into something like `Qwerwer Wwere`. User reads that line in app about `Qwerwer Wwere` and thinks — hell what it is? He will try to search for that term, but since translator used not commonly used proper translation, user won't find anything.

Though, I'm about 90% ensured that in fact I'm digging into problem, that not related particularly to that issue. Unfortunately, not much help here I can provide, since I don't know how important in Chinese phrasing, used words and so on. For example, Russian in compare with English has much more difficult and important structure. You can change just few little bits, use proper, but a bit not fitting word variant, and whole phrase will sound just wrong.

Anyway, as I've already mentioned, it's completely up to you. It's good to improve translation, but we always should keep improvements sane and normalized.

**Comment by LNDDYL** [comment]
我再重複一遍：「,」是逗點（逗號）。

**Comment by petercpg** [comment]
@ArmorDarks I think you are digging into exactly the problem. I'm more confident now that using a Github-flow is better as I'm trying not to be dictating, allowing contributors discuss to dissolve such conflicts. Thanks again for your understanding :)

@LNDDYL 請見以下英文逗號與中文頓號的用法，如果上面的圖片還看不出來我想這應該更清楚了。
* http://en.wikipedia.org/wiki/Comma#In_lists
* http://en.wikipedia.org/wiki/Comma#In_other_languages
* http://www.edu.tw/FILES/SITE_CONTENT/M0001/HAU/h2.htm

**Comment by ArmorDarks** [comment]
@petercpg Sure. That's is what Pull Requests for :)

**Comment by ArmorDarks** [comment]
So, are there any progress on this? :) Should we merge it partially?

**Comment by petercpg** [comment]
Seems no more feedback was received and we can move on. Let me figure out which way is better to merge only some lines.

**Comment by ArmorDarks** [comment]
Sure. Maybe just manual picking of correct lines will be quicker.

**Comment by petercpg** [comment]
Commits pushed, closing this issue.

**Comment by ArmorDarks** [comment]
Great, thanks!

**Pull request closed without merging**